nil/principles.mdx

```diff
@@ -0,0 +1,64 @@
+import Flows from '@site/static/img/nil/flows.png'
+
+# Principles
+
+This page offers a high-level explanation of the architecture of =nil;.
+
+## Flows
+
+The following diagram shows how =nil; organizes execution shards, the main shard, the transfer of ZKPs, and communications with Ethereum.
+
+<img src={Flows}/ >
+
+In brief, each execution shard is responsible for generating a ZKP for their respective state change. Then the proofs are verified via a [**HotStuff2-based**](https://eprint.iacr.org/2023/397.pdf) global consensus protocol. 
```

Comment:
It's strange to emphasize hotstuff-based protocol only for consensus shard, when it's also true for execution shards

Reply:
Fair, I've added info about local consensus too